sips/sip-025/sip-025-sbtc.md

```diff
@@ -0,0 +1,907 @@
+
+There is no transaction fee for the Stacker authorization transaction.  Stackers
+instead require the miner to include it for free in their blocks by withholding
+block signatures until it is included.
+
+### Failure Modes and Recovery
+
+There are several places where a deposit can fail.  First, the Stackers may fail
+to complete a signing round that achieves the 70% threshold.  One of the
+following outcomes will happen:
+
+* The Stackers produce a deposit authorization with at least 70% of the STX
+  voting against it.  In this case, the sBTC is not minted, and the user can
+later reclaim their BTC once the subsequent reward cycle finishes (i.e. the
+reward cycle _after_ this current reward cycle).
+
+* The Stackers are unable to come to agreement on "yes" or "no".  In this case,
+  the Stackers record this outcome by submitting each Stacker's individual votes as
+a bitmap to the `.sbtc` contract.  This is a vote to abstain.  No sBTC will be
+minted, but a record of how each Stacker voted is preserved.  The user later
+reclaims their BTC once the subsequent reward cycle finishes.
+
+* A new set of Stackers come online as part of a new reward cycle starting while
+  the deposit request is pending.  In this case, the deposit request is cancelled
+-- the new Stackers cannot act on it.  The spending conditions of the
+deposit script will permit the user to reclaim their Bitcoin later.
+
+* The Stackers are unable to complete a WSTS signing round.  In this case, the
+  sBTC system freezes until the Stackers are able to come back online -- no
+deposits or withdraws will be processed.  Freezing
+and thawing the system are discussed in a subsequent section.
+
+Second, the miner can fail to include the deposit authorization transaction from
+the Stackers.  This would prevent the Stackers' vote from being recorded in
+`.sbtc`.  If this happens, then Stackers refuse to sign any blocks until they
+see a block which contains the as-yet-unmined authorization transaction(s).  If
+need be, Stackers themselves can spend some BTC to become miners in order to
+push their transactions through.
+
+### Transaction Fees
+
+The user pays the transaction fee for their sBTC deposit transaction.  Stackers
+do not pay a STX transaction fee, but instead require miners to include their
+deposit authorization transaction in the next block they sign.
+
+### Two-phase Deposit
+
+Users may not be able to send a Bitcoin transaction with an `OP_RETURN` output.
+For example, this is difficult to do or is unsupported in some Bitcoin wallets,
+especially custodial wallets.  To support these users, a 3rd party could simply
+send deposit transaction on their behalf.  The user would simply pay the 3rd
+party the requisite Bitcoin to deposit, and the 3rd party would generate and send
+the deposit transaction for them.
+
+## Withdraw
+
+Withdrawing is the act of redeeming sBTC for BTC (Figure 2).
+To redeem sBTC for BTC, the user first sends a Stacks trnsaction to `.sbtc`
```

Comment:
The current Alpha Romeo 0.1 developer release has implemented the withdrawal as a btc transaction. I suggest to add the btc withdrawal tx as a second option?

Reply:
Sure, this can be done.